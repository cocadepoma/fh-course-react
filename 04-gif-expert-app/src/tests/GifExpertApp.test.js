import React from 'react';
import {shallow} from 'enzyme'
import GiftExpertApp from '../GifExpertApp';

describe('pruebas en <GifExpertApp/>', () => {

    test('comprobar que el componente se muestra correctamente', () => {
        
        const wrapper = shallow(<GiftExpertApp/>);

        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar una lista de categorias', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GiftExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GiftGrid').length).toBe(categories.length);

    });
    
    



});
