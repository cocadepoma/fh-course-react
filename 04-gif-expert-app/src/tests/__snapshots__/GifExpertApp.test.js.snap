// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp/> comprobar que el componente se muestra correctamente 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`pruebas en <GifExpertApp/> debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiftGrid
      category="One Punch"
      key="One Punch"
    />
    <GiftGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;
